saga/sagaCoordinator.go

```diff
@@ -0,0 +1,79 @@
+package saga
+
+//
+// Saga Object which provides all Saga Functionality
+// Implementations of SagaLog should provide a factory method
+// which returns a saga based on its implementation.
+//
+type SagaCoordinator struct {
+	log SagaLog
+}
+
+//
+// Make a Saga which uses the specied SagaLog interface for durable storage
+//
+func MakeSagaCoordinator(log SagaLog) SagaCoordinator {
+	return SagaCoordinator{
+		log: log,
+	}
+}
+
+// Make a Saga add it to the SagaCoordinator, if a Saga Already exists
+// with the same id, it will overwrite the already existing one.
```

Comment:
Can we flesh out the overwriting part of this and why that's ok vs just panicking? Doesn't that mean there could be multiple goroutines running different jobs one succeeds, the other fails and can't do rollback because the saga is already done?